var/spack/repos/builtin/packages/scotch/package.py

```diff
@@ -171,20 +158,34 @@ def configure(self):
                 fh.write('\n'.join(makefile_inc))
 
     def install(self, spec, prefix):
-        self.configure()
-
         targets = ['scotch']
         if '+mpi' in self.spec:
             targets.append('ptscotch')
 
-        if '+esmumps' in self.spec:
-            targets.append('esmumps')
-            if '+mpi' in self.spec:
-                targets.append('ptesmumps')
+        if self.spec.version >= Version('6.0.0'):
```

Comment:
the change of file makes it work. 
for version  < 6.0.0 the target does not exists but the file_esmumps always build the library and the file without _esmumps never build it.  si the chamge in url_for_version is sufficient and so no need to patch the makefiles